contracts/RollupPoB.sol

```diff
@@ -0,0 +1,331 @@
+        address bonusAddress,
+        bool useBonus
+    ) external payable {
+        require(slot >= currentSlot() + minNumSlots, 'This auction is already closed');
+        doBid(slot, beneficiaryAddress, forgerAddress, withdrawAddress, bonusAddress, useBonus);
+    }
+
+    /**
+     * @dev distribution of the amount
+     * @param amount amount to distribute
+     * @param op operator who will receive the amount
+     */
+    function _returnBid(uint amount, uint bonus, Operator storage op) private {
+        withdrawBid[op.withdrawAddress] += amount;
+        bonusBalance[op.bonusAddress] += bonus;
+    }
+    /**
+     * @dev function to withdraw bid
+     */
+    function withdraw() external {
+        require(withdrawBid[msg.sender] > 0, 'You cannot withdraw the amount');
+        uint auxAmount = withdrawBid[msg.sender];
+        withdrawBid[msg.sender] = 0;
+        msg.sender.transfer(auxAmount);
+    }
+
+    /**
+     * @dev operator commits data and forge a batch
+     * @param compressedTx data committed by the operator. Represents off-chain transactions
+     * @param proofA zk-snark input
+     * @param proofB zk-snark input
+     * @param proofC zk-snark input
+     * @param input public zk-snark inputs
+     */
+    function commitAndForge(
+        bytes calldata compressedTx,
+        uint[2] calldata proofA,
+        uint[2][2] calldata proofB,
+        uint[2] calldata proofC,
+        uint[8] calldata input
+    ) external virtual {
+        uint32 slot = currentSlot();
+        Operator storage op = slotWinner[slot];
+        // message sender must be the controller address
+        require(msg.sender == op.forgerAddress, 'message sender must be forgerAddress');
+        // Check if deadline has been achieved to not commit any more data
+        uint blockDeadline = getBlockBySlot(slot + 1) - SLOT_DEADLINE;
+        require(getBlockNumber() < blockDeadline, 'not possible to commit data after deadline');
+        uint256 offChainHash = hashOffChainTx(compressedTx, MAX_TX);
+        emit dataCommitted(offChainHash);
+        // Check input off-chain hash matches hash commited
+        require(offChainHash == input[offChainHashInput],
+            'hash off chain input does not match hash commited');
+        rollupInterface.forgeBatch(op.beneficiaryAddress, proofA, proofB, proofC, input);
+        // one block has been forged in this slot
+        infoSlot[slot].fullFilled = true;
+    }
+
+     /**
+     * @dev commitAndForge after deadline
+     * @param compressedTx data committed by the operator. Represents off-chain transactions
+     * @param proofA zk-snark input
+     * @param proofB zk-snark input
+     * @param proofC zk-snark input
+     * @param input public zk-snark inputs
+     */
+    function commitAndForgeDeadline(
+        bytes calldata compressedTx,
+        uint[2] calldata proofA,
+        uint[2][2] calldata proofB,
+        uint[2] calldata proofC,
+        uint[8] calldata input
```

Comment:
- input should be `uint[10]` and it should match with `RollupInterface.sol`
- Applies to `commitAndForgeDeadline` function as well